src/main/java/jimmy/mcgymmy/model/food/Macronutrient.java

```diff
@@ -5,7 +5,7 @@
 
 public abstract class Macronutrient {
     public static final String MESSAGE_CONSTRAINTS = "values should only contain positive numbers";
-    private static final String VALIDATION_REGEX = "(\\d)+";
+    private static final String VALIDATION_REGEX = "(\\d){1,3}";
```

Comment:
Alternatively if we wanted to cap it at 5000 we can do "[012345]?(\\d){1,3}" or sth